### PR TITLE
disable warts.Any and remove most suppressions

### LINKED ISSUE #6116: Any wart has too many false positives in Scala code
The `Any` wart has multiple false positives that are pervasive through our code. Therefore, we should disable it.

The false positives fall into a few categories.

The first category is caused by various aspects of Scalatest. The ironic outcome of this is that we substitute safe constructs like `xs shouldBe empty` for unsafe constructs like `xs shouldBe 'empty` because the latter don't trigger the `Any` wart, even though the _former_ are well-typed and the latter are not.

The second category is wartremover/wartremover#463. This is the most pervasive, and unlike the prior category can't be fixed by simply limiting the wart to main code. In fact, even code that merely _infers_ the values of higher-kinded type parameters can trigger `Any`.

The third category is the standard collections `view` construct. Again, we use it as a refactoring-safe, 2.13-safe variant of `iterator`, as in #6075. But since `view` triggers `Any` in some cases but `iterator` does not, we ironically choose the latter, less safe construct.

I have confirmed that Wartremover 2.4.8 does not fix any of these categories of false positives.

I suspect that the vast majority, albeit not all, of the 229 `warts.Any` suppressions in our codebase fall into one of these three categories. I don't suggest disabling a safety-enhancing check lightly, but make an exception in this case, as we are counterproductively drawn to less type-safe constructs as described above, as things stand. The goal is to allow us to use more safe constructs conveniently, not to expand our usage of unsafe Scala constructs (i.e. uses of `Any` "in anger" that are actually unsafe).

I _only_ suggest disabling the `Any` wart. The other warts are quite good.

### PR DESCRIPTION
**Pull request by S11001001**
This globally disables the `warts.Any` wart and removes almost all of its suppressions.

I have left in `warts.Any` suppressions that appear to call out genuine cases where we use `Any`, so that reintroduction of the checker will be easier should the false positives in #6116 be fixed in a wartremover release >2.4.8. These suppressions are currently no-op, but there was never a way to tell whether they had become no-op anyway, and I do not want to place needless barriers in the way of reintroducing a hypothetical fixed `warts.Any`.

For example, this suppression is preserved; it calls out a deliberate but non-greppable use of `Any`, so we would not want it to be warned about should we reintroduce `warts.Any`.

https://github.com/digital-asset/daml/blob/a2d785e3ee29c56b2b06780897fc7407666767f8/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SomeArrayEquals.scala#L15-L26

At 4b067a2c5eea36fecdea0f5d543c3206f41c19b3 , 29 such suppressions remain, 200 having been removed.

Closes #6116.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: bazel_tools/scala.bzl [hunk 1]
```diff
@@ -72,7 +72,7 @@ plugin_scalacopts = [
     "-Xplugin-require:wartremover",
 
     # This lists all wartremover linting passes.
-    "-P:wartremover:traverser:org.wartremover.warts.Any",
+    # "-P:wartremover:traverser:org.wartremover.warts.Any",
```
Comment (S11001001) [inline bazel_tools/scala.bzl:75]: Commenting out is proper here; note the context.

File: ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/sandbox/ScenarioLoadingITBase.scala [hunk 2]
```diff
@@ -30,7 +30,6 @@ import scala.concurrent.Future
 
 @SuppressWarnings(
```
Comment (SamirTalwar-DA) [inline ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/sandbox/ScenarioLoadingITBase.scala:31]: Can you join these into one line where possible?

File: ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/ScenarioLoadingITDivulgence.scala [hunk 3]
```diff
@@ -17,7 +17,6 @@ import org.scalatest.{Matchers, WordSpec}
 
 @SuppressWarnings(
```
Comment (SamirTalwar-DA) [inline ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/ScenarioLoadingITDivulgence.scala:18]: Probably here too.

Comment (S11001001) [reply ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/ScenarioLoadingITDivulgence.scala]: 👍 536eb66462

**Pull request merged**